code/modules/projectiles/ammunition/ballistic/pistol.dm

```diff
@@ -76,3 +78,15 @@
 	. = ..()
 	if(!isturf(target))
 		loaded_projectile.set_homing_target(target)
+		new /obj/effect/temp_visual/smartgun_target(get_turf(target))
+	else
+		var/atom/aimbot_target = locate(/mob/living) in range(auto_lock_range, target)
+		if(aimbot_target)
+			loaded_projectile.set_homing_target(aimbot_target)
+			new /obj/effect/temp_visual/smartgun_target(get_turf(aimbot_target))
```

Comment:
I think that you should generalize this behavior (either as an element or on casings themselves) and give it to heartseekers and psyker revolvers (the one burdened chaplain gets) - their homing is useless for the same reason. Lionhunters are fine as they have long range and full piercing, so homing is more of an insurance policy for them, but for former two homing is absolutely useless in their current state. Maybe give them 1 tile of homing instead of 2 as to compensate for their higher damage.

Reply:
how it feels to be out of scope

Reply:
Still would prefer this to be separated from casings for the afformentioned reason, not insisting that you should buff other guns in the same PR